app/src/test/java/com/android/sample/model/profile/MockProfilesRepository.kt

```diff
@@ -0,0 +1,89 @@
+package com.android.sample.model.profile
+
+class MockProfilesRepository : ProfilesRepository {
+
+  private val userProfiles =
+      mutableMapOf<String, User>(
+          "u1" to
+              User(
+                  id = "u1",
+                  name = "Alice",
+                  surname = "Smith",
+                  interests = listOf("Hiking", "Cycling"),
+                  activities = listOf("a1", "a2"),
+                  photo = null,
+                  likedActivities = listOf("a1")))
+  private val activitiesList =
+      mutableMapOf<String, MutableList<String>>("u1" to mutableListOf("a1", "a2"))
+
+  override fun getUser(userId: String, onSuccess: (User?) -> Unit, onFailure: (Exception) -> Unit) {
+    try {
+      // Retrieve the user profile by userId
+      val user = userProfiles[userId]
+      onSuccess(user)
+    } catch (e: Exception) {
+      onFailure(e)
+    }
+  }
+
+  override fun addActivity(
+      userId: String,
+      activityId: String,
+      onSuccess: () -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    try {
+      // Add activity to user's activity list
+      if (userProfiles.containsKey(userId)) {
+        activitiesList.computeIfAbsent(userId) { mutableListOf() }.add(activityId)
+        onSuccess()
+      } else {
+        throw Exception("User not found")
+      }
+    } catch (e: Exception) {
+      onFailure(e)
+    }
+  }
+
+  override fun addLikedActivity(
+      userId: String,
+      activityId: String,
+      onSuccess: () -> Unit,
+      onFailure: (Exception) -> Unit
+  ) {
+    TODO("Not yet implemented")
+  }
```

Comment:
Is it someone else task to do this? If yes, maybe mention it on your PR description